### ISSUE TITLE
Wanted: Command line option to start testing partway through a suite

### ISSUE THREAD

**Issue by milhead2** [robotframework/robotframework#3446]
If restarting midpoint through a long test sequence it would be nice to have a command line option (lets call it --dryrununtil) that you pass a matching test string to (like --test) that would allow me to 
1) Run a set of tests with our complex command line
2) Have it fail at a point (something like "5.3.1.2 UDS Reflash Verify")
3) Ctrl-C out to terminate the run and fix the error then restart the sequence with 
    robot ...normal complex variable/include/excludes... --dryrununtil "5.1.3.2*" ./all_tests
4) The goal of this command is to skip the tests until the one matching --dryrununtil is encountered.. The matching and all subsequent tests are then run.

**Comment by pekkaklarck** [comment]
Having a command line option to start execution from a certain test sounds like a useful feature to me. Would probably work nice also with the `--ExitOnFailure` option.

I don't think switching execution mode during execution would a good idea, though. It would make it harder to merge results of partially run suites and implementation would also be really hard. Just dropping tests from the executed suite would, in my opinion, work a lot better. As a pretty big bonus, it would also be very easy to implement. Would you @milhead2 be interested to look at that? I can help if needed but probably won't have time to implement this myself for RF 3.2.

**Comment by pekkaklarck** [comment]
I think adding an option like `--startfrom` or possibly `--startfromtest` would work pretty well. Could also consider adding `--endto` (or `--endtotest`). In combination they would allow running suites in smaller batches either in sequence or in parallel.

**Comment by milhead2** [comment]
I'll take a look at it but it won't be immediate, I've not dipped my toe into the robotframework pool except as a user..  I understand that means that I should get my toes/feet wet (grin), I'll let you know how it goes.

**Comment by milhead2** [comment]
Is this along the lines of what you are thinking? 
The way I've drafted the option makes the start inclusive..  My gut feeling is that the end 
would be exclusive but my first description below indicates inclusive..   Your thoughts?
```
    --startfromtest name *  Will skip tests prior to the first occurrence of 
                          a test that matches 'name'. The matching test is 
                          inclusive and will be executed. (see the --test 
                          option for a description of name matching)
    --endwithtest name *  Will skip all tests after the first occurrence of a 
                          test that  matches 'name' (see the --test option for 
                          a description of name matching)
```

**Comment by pekkaklarck** [comment]
Yeah, something like that. I also think start should be inclusive and end exclusive. That allows something like

    robot --endwithtest Example tests.robot
    robot --startfromtest Example tests.robot

without the Example tests being executed twice.

How's the option naming? For me "start from" and "end to" would be logical, but I'm not a native English speaker and don't know would "end with" be grammatically better. Would "start with" and "end with" work?

**Comment by pekkaklarck** [comment]
Technically this should be implemented using pre-run modifiers. Implementation ought to be pretty simple, but we'd obviously need to add some tests as well. The first prototypes could be done simply using the `--prerunmodifier` option without touching Robot's internals at all.

**Comment by pekkaklarck** [comment]
This has been stale for five years. Can be reopened, or new issue submitted, if ther need arises again.
